test/controllers/tasks_controller_test.rb

```diff
@@ -59,70 +76,161 @@
     end
   end
 
-  describe "create" do
-    it "can create a new task" do
-      skip
-
+  describe 'create' do
+    it 'can create a new task' do
       # Arrange
+      # Note to students:  Your Task model **may** be different and
+      #   you may need to modify this.
       task_hash = {
         task: {
-          name: "new task",
-          description: "new task description",
-          completion_date: nil,
-        },
+          name: 'new task',
+          description: 'new task description',
+          completion_date: nil
+        }
       }
 
       # Act-Assert
       expect {
         post tasks_path, params: task_hash
-      }.must_change "Task.count", 1
+      }.must_change 'Task.count', 1
 
       new_task = Task.find_by(name: task_hash[:task][:name])
       expect(new_task.description).must_equal task_hash[:task][:description]
-      expect(new_task.due_date.to_time.to_i).must_equal task_hash[:task][:due_date].to_i
-      expect(new_task.completed).must_equal task_hash[:task][:completed]
+      expect(new_task.completion_date).must_equal task_hash[:task][:completion_date]
 
       must_respond_with :redirect
-      must_redirect_to task_path(new_task.id)
+      must_redirect_to tasks_path
     end
   end
 
   # Unskip and complete these tests for Wave 3
-  describe "edit" do
-    it "can get the edit page for an existing task" do
-      skip
-      # Your code here
+  describe 'edit' do
+    it 'can get the edit page for an existing task' do
+      test_task = task
+
+      get edit_task_path(task.id)
+
+      must_respond_with :ok
     end
 
-    it "will respond with redirect when attempting to edit a nonexistant task" do
-      skip
-      # Your code here
+    it 'will respond with redirect when attempting to edit a nonexistant task' do
+      get edit_task_path(-1)
+
+      must_respond_with :redirect
+      must_redirect_to tasks_path
     end
   end
 
   # Uncomment and complete these tests for Wave 3
-  describe "update" do
+  describe 'update' do
     # Note:  If there was a way to fail to save the changes to a task, that would be a great
     #        thing to test.
-    it "can update an existing task" do
-      skip
-      # Your code here
+    it 'can update an existing task' do
+      test_task = task
+      task_hash = {
+        task: {
+          name: 'new task',
+          description: 'new task description',
+          completion_date: nil,
+          completed: true
+        }
+      }
+
+      # Act-Assert
+      expect {
+        patch task_path(test_task), params: task_hash
+      }.wont_change 'Task.count'
+
```

Comment:
You should also test that the tasks' fields are changed to match the hash above.